### DISCUSSION TITLE
Eclair on Signet

### DISCUSSION THREAD

**Discussion by ShahanaFarooqui** [ACINQ/eclair#2382] [Q&A] [answered]
Hi, do you have any plans on making Eclair available on Signet? 

LND and Core Lightning both are compatible with signet and it will be helpful in testing updates if Eclair can also run on Signet. Another solution is to run all of them on Regtest which I am doing currently but it requires manual block generation each time.

**Comment by remyers** [comment, marked as answer]
To do this cleanly we'll need to first update bitcoin-kmp  to define the signet genesis block  in [PR #61](https://github.com/ACINQ/bitcoin-kmp/pull/61) and bitcoin-lib to expose it in [PR #70](https://github.com/ACINQ/bitcoin-lib/pull/70) so Eclair can access it.

**Reply by remyers** [reply]
You can help with the Signet version of Eclair by reviewing and/or testing [PR #2387](https://github.com/ACINQ/eclair/pull/2387). Let me know if you have any questions!

**Comment by ShahanaFarooqui** [comment]
@remyers and @t-bast Thank you for the quick PR. 

I am getting below error while testing this merge:
`fatal error: cannot retrieve feerates, channels may be at risk: ensure bitcoind estimatesmartfee correctly returns feerates and restart eclair`

I confirm that my bitcoind.conf file sets `fallbackfee=0.0002`.

**Reply by remyers** [reply]
Hmm, that's strange. Can you share with me your bitcoind.conf and eclair.conf files so I can try to reproduce the issue? Thanks!

Do you see a problem if you run `bitcoin-cli estimatesmartfee 1088`, for example ?

**Reply by ShahanaFarooqui** [reply]
It is working after I restarted my machine today. I changed nothing in my setup.

`bitcoin-cli -rpcport=8334 -rpcuser=* -rpcpassword=** estimatesmartfee 1088`
`error code: -8
error message:
Invalid conf_target, must be between 1 and 1008`

`bitcoin-cli -rpcport=8334 -rpcuser=* -rpcpassword=** estimatesmartfee 1008`
`
{
  "feerate": 0.00001000,
  "blocks": 22
}`

`ecl-cli -p *** -a http://localhost:9090 getinfo` is also returning node information.

Sharing my bitcoin.conf and eclair.conf files though the setup is working fine now. 

bitcoin.conf:
===========================================
signet=1
server=1
rpcuser=*
rpcpassword=**
rpchost=127.0.0.1
rpcworkqueue=128
rpcclienttimeout=30
fallbackfee=0.0002
zmqpubrawblock=tcp://127.0.0.1:38332
zmqpubrawtx=tcp://127.0.0.1:38333
dbcache=6000
prune=0
txindex=1

[signet]
rpcport=8334


eclair.conf
===========================================
eclair.server.port=9737
eclair.node-alias="ECLSignet1"
eclair.node-color=D94BBB
eclair.trampoline-payments-enable=true

eclair.api.enabled=true
eclair.api.password=***
eclair.api.port=9090
eclair.chain=signet

eclair.bitcoind.rpcport=8334
eclair.bitcoind.rpcuser=*
eclair.bitcoind.rpcpassword=**
eclair.bitcoind.zmqblock="tcp://127.0.0.1:38332"
eclair.bitcoind.zmqtx="tcp://127.0.0.1:38333"


Thank you once again for the quick PR.

**Reply by remyers** [reply]
Thanks for the update, I don't see anything suspicious in your configs either. Please do let us know if you see this problem again, and I'll also keep my eye out for it as I work more with Signet.
